tst/regression/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+#=========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+# Adding regression tests
+#
+# To add additional regression tests to ctest, to things must be added to 
+# this file:
+# 
+# 1. there must exist within the regression/test_suites directory a folder associated with the test
+#   * For the calculate_pi example the folder is called 'calculate_pi', This
+#     folder must be added to the cmake variable TEST_DIRS as shown below. 
+
+# 2. the correct arguments to pass to the 'run_test.py' script must be specfied
+#   * This arguments must be added to the CMAKE variable TEST_ARGS as shown below
+#
+# Note each new test must append exactly 1 item to each of the CMAKE variables
+
+function(setup_test dir arg)
+  separate_arguments(arg) 
+  add_test( NAME regression_test:${dir} COMMAND python "${CMAKE_CURRENT_SOURCE_DIR}/run_test.py" ${arg} --test_dir "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/${dir}")
+  set_tests_properties(regression_test:${dir} PROPERTIES LABELS "regression" )
+endfunction()
+
+function(setup_test_mpi dir arg)
+  if( MPI_FOUND )
+    separate_arguments(arg) 
+    add_test( NAME regression_mpi_test:${dir} COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/run_test.py  --mpirun ${MPIEXEC_EXECUTABLE} --mpirun_opts "${MPIEXEC_NUMPROC_FLAG} ${MPIEXEC_MAX_NUMPROCS} ${MPIEXEC_PREFLAGS}" ${arg} --test_dir ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/${dir})
+    set_tests_properties(regression_mpi_test:${dir} PROPERTIES LABELS "regression,mpi" RUN_SERIAL ON )
+  else()
+    message(STATUS "MPI not found, not building regression tests with mpi")
+  endif()
+endfunction()
+
+list(APPEND TEST_DIRS calculate_pi)
+list(APPEND TEST_ARGS "--driver ${CMAKE_BINARY_DIR}/example/calculate_pi/pi-example --driver_input ${CMAKE_CURRENT_SOURCE_DIR}/test_suites/calculate_pi/parthinput.regression")
```

Comment:
I'm not thrilled that these are lists, making it a little hard to map dirgs to args when debugging. But I think that's just cmake and we should leave it.